src/ucp/tag/rndv.c

```diff
@@ -66,6 +96,59 @@ size_t ucp_tag_rndv_pack_rkey(ucp_request_t *sreq, ucp_lane_index_t lane,
     return 0;
 }
 
+static size_t ucp_tag_rndv_pack_mrail_rkeys(ucp_request_t *sreq, void *rkey_buf, uint16_t *flags)
+{
+    ucp_ep_t        *ep      = sreq->send.ep;
+    ucp_dt_state_t  *state   = &sreq->send.state.dt;
+    size_t           packet  = 0;
+    int              cnt     = 0;
+    int              i;
+    ucp_lane_index_t lane;
+
+    ucs_assert(UCP_DT_IS_CONTIG(sreq->send.datatype));
+
+    cnt = ucp_request_mrail_reg(sreq);
+    ucs_assert_always(cnt <= UCP_MAX_RAILS);
+
+    if (cnt) {
+        for (i = 0; i < cnt && i < UCP_MAX_RAILS; i++) {
+            ucs_assert_always(!ucp_request_is_empty_rail(state, i));
+
+            lane = state->dt.mrail[i].lane;
+
+            UCS_PROFILE_CALL(uct_md_mkey_pack, ucp_ep_md(ep, lane),
+                             state->dt.mrail[i].memh, rkey_buf + packet);
+            packet += ucp_ep_md_attr(ep, lane)->rkey_packed_size;
```

Comment:
we need some check that we are not exceeding max_bcopy or max_short (whichever is used to send RTS)

Reply:
added assert